packages/thirdweb/src/react-ui/providers/thirdweb-provider.tsx

```diff
@@ -0,0 +1,157 @@
+import { en } from "../ui/locales/en.js";
+import { LazyConnectModal } from "../ui/ConnectWallet/Modal/LazyConnectModal.js";
+import type { ThirdwebProviderProps as ThirdwebProviderPropsReact } from "../../react/providers/thirdweb-provider.js";
+import { ThirdwebProvider as ThirdwebProviderReact } from "../../react/providers/thirdweb-provider.js";
+import type { ThirdwebLocale } from "../ui/locales/types.js";
+import { defaultWallets } from "../wallets/defaultWallets.js";
+import type { WalletConfig } from "../../react/types/wallets.js";
+import { WalletUIStatesProvider } from "./wallet-ui-states-provider.js";
+import { ThirdwebLocaleContext } from "./locale-provider.js";
+
+/**
+ * The ThirdwebProvider is component is a provider component that sets up the React Query client and Wallet Connection Manager.
+ * To you thirdweb React SDK's hooks and components, you have to wrap your App component in a ThirdwebProvider.
+ *
+ * `ThirdwebProvider` requires a `client` prop which you can create using the `createThirdwebClient` function.  You must provide a `clientId` or `secretKey` in order to initialize a `client`.
+ * You can create an Api key for free at from the [Thirdweb Dashboard](https://thirdweb.com/create-api-key).
+ * @param props - The props for the ThirdwebProvider
+ * @example
+ * ```jsx
+ * import { createThirdwebClient } from "thirdweb";
+ * import { ThirdwebProvider } from "thirdweb/react";
+ *
+ * const client = createThirdwebClient({
+ *  clientId: "<your_client_id>",
+ * })
+ *
+ * function Example() {
+ *  return (
+ *    <ThirdwebProvider client={client}>
+ *      <App />
+ *    </ThirdwebProvider>
+ *   )
+ * }
+ * ```
+ * @component
+ */
+export function ThirdwebProvider(props: ThirdwebProviderProps) {
```

Comment:
how come this is in /react-ui ? i thought we said all providers + hooks would be shared?

Reply:
The main thirdweb-provider is in the `react` folder. This ThirdwebProvider is specific to react for two reasons: 

1. We don't share the same locale type for web vs mobile: I want to merge them and have something like:
```
const locale = {
   web: {}, // will be empty for mobile
   mobile: {}, will be empty for web,
   common: {}
}
```
Issue here is that there's no tree-shaking but there's not too much overhead if leave mobile empty.

The main pushback here is that it may be confusing for devs when want to customize the strings to find this mobile object in React for example.

2. We need to be able to pass defaultWallets which is also different: this can be fixed by a getDefaultWallets